content/actions/creating-actions/creating-a-docker-container-action.md

```diff
@@ -47,13 +47,16 @@ Before you begin, you'll need to create a GitHub repository.
 In your new `hello-world-docker-action` directory, create a new `Dockerfile` file. For more information, see "[Dockerfile support for {% data variables.product.prodname_actions %}](/actions/creating-actions/dockerfile-support-for-github-actions)."
 
 **Dockerfile**
-```dockerfile
+```dockerfile{:copy}
 # Container image that runs your code
 FROM alpine:3.10
 
 # Copies your code file from your action repository to the filesystem path `/` of the container
 COPY entrypoint.sh /entrypoint.sh
 
+# Makes your code file executable
+RUN chmod +x entrypoint.sh
+
```

Comment:
@avinal 👋 
Would you be able to change this to the original `chmod +x entrypoint.sh` version? The feedback I've received is that this approach adds another layer to the container, which would affect pull performance.

Reply:
Yeah, this adds another layer to the container. But as much I have used Dockerized GitHub Action, the script will not run without permission, this is also evident from the issue comment https://github.com/github/docs/issues/974#issue-734169633 for which this PR was submitted. You may want to see into that matter again. 

I can suggest an improvement to this, we can combine this line with other commands using `&&` such that it gets executed without an extra layer.

Reply:
@avinal - sorry about the delay here 🙇  It looks like the approach in the existing documentation is to run `chmod +x entrypoint.sh` separately: (step 2 here) https://docs.github.com/en/free-pro-team@latest/actions/creating-actions/creating-a-docker-container-action#writing-the-action-code. Was this step not working for you?

Reply:
@martin389 Sorry for the delay, In the step two it is mentioned that run this command to make your file executable, but it is not mentioned that where you are supposed to run `chmod +x entrypoint.sh`, is it our system or inside docker? And that is the point of whole confusion. Secondly I am unsure about if the file will remain executable inside docker too. 

Secondly I had to add this line before running the entrypoint.sh file(maybe because I didn't already made them executable, no body does !!) Please give me 1 hour and I will test the exact steps described in the docs.

Reply:
Thanks @avinal - the intention is for `chmod +x entrypoint.sh` to run on the user's system, and not within Docker. I agree that this is something that could be clearer 👍 

I tested these steps yesterday on macOS and the script remained executable, but I'd be interested to know if it works for you too. Thanks for all your help here! 🎊

Reply:
Here are the test cases I did
- Not making them executable in system - NO(expected)
- Making them executable in system using `chmod +x entrypoint.sh`- NO(not expected)
- Making them executable in system using `git update-index --chmod=+x entrypoint.sh` - NO(not expected)
- Making the system same (both ubuntu:latest) - NO(not expected)
- Adding the specific line in Dockerfile - YES(as usual)

Surprisingly the example works but the exact copy doesn't. I am pretty confused now😅

Reply:
That would be great, thank you! 🎊